syntax = "proto3";

import "google/protobuf/any.proto";

package fcpb.p2p.v1.auth;

/**
 * Request to authenticate as an active validator
 */
message AuthenticateSession {
    // The maximum understood protocol version of the client
    int32 version = 1;
    // A random 8 byte identifier chosen by the client
    bytes client_session_id = 2;
    // Validator authentication stake public key address
    string auth_address = 3;
}

/**
 * Response that establishes an authenticated session
 */
message AuthenticateSessionResponse {
    // The maximum understood protocol version of the server
    int32 version = 1;
    // A random 8 byte identifier chosen by the server
    bytes server_session_id = 2;
}
